src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -51,48 +53,118 @@ public boolean isMaintainer() {
     return this.isMaintainer;
   }
 
-  public static GivrUser getUserByIdFromDatastore(String userId) {
+  public String getUserEmail() {
+    return this.email;
+  }
+
+  public boolean isLoggedIn() {
+    return this.isLoggedIn;
+  }
+
+  public static PreparedQuery getPreparedQueryResultUserWithProperty(String propertyName, String propertyValue) {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Filter queryFilter = new FilterPredicate("userId", FilterOperator.EQUAL, userId);
+
+    Filter queryFilter = null;
+    if (propertyName.equals("isMaintainer")) {
+      queryFilter = new FilterPredicate(propertyName, FilterOperator.EQUAL, Boolean.parseBoolean(propertyValue));
+    } else {
+      queryFilter = new FilterPredicate(propertyName, FilterOperator.EQUAL, propertyValue);
+    }
     Query query = new Query("User").setFilter(queryFilter);
-      
-    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(1);
+
     PreparedQuery preparedQuery = datastore.prepare(query);
+    return preparedQuery;
+  }
+
+  // Check if User with propertyName, propertyValue exists within Datastore.
+  public static boolean checkIfUserWithPropertyExists(String propertyName, String propertyValue) {
```

Comment:
I would simplify this to ;

private static boolean ensureSingleResult(QueryResultList<Entity> userResult).

That way you only make the request once and use the same result list for getting the user as well as this check. Any reason to make the function public ?

Reply:
This function currently is used in AddMaintainerServlet when checking if the new Maintainer exists within the Datastore. It's also used in the AuthenticateServlet to check if the Id exists; if not, then it checks if the email exists.